app/iot/service.py

```diff
@@ -18,32 +18,54 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
+        attempts_left = 7
+        while device_id in self.devices:
+            print(
+                f"Device with id {device_id} already registered"
+                f"Gererating new one..."
+            )
+            device_id = generate_id()
+            attempts_left -= 1
+            if attempts_left == 0:
+                print("Unable to register device, trying again...")
+            print(f"Attempts left: {attempts_left}")
```

Comment:
Code style issue: This print statement should be inside the if block. Otherwise, it will print the number of attempts left every time the loop iterates, even if the device_id is unique.